azurelinuxagent/ga/persist_firewall_rules.py

```diff
@@ -176,6 +182,15 @@ def _setup_network_setup_service(self):
         # This is to handle the case where WireIP can change midway on service restarts.
         # Additionally, incase of auto-update this would also update the location of the new EGG file ensuring that
         # the service is always run from the most latest agent.
+        osutil = get_osutil()
+        # If RHEL and in image mode, we need to clean up the service file in old path
+        if self._DISTRO == 'rhel' and osutil.is_image_mode() and os.path.exists(self.get_service_file_path(mode="package_mode")):
+            logger.info("Cleaning up old service file in image mode: {0}".format(self.get_service_file_path(mode="package_mode")))
+            try:
+                fileutil.rm_files(self.get_service_file_path(mode="package_mode"))
```

Comment:
The cleanup would fail since image mode is read-only, wouldn't it?

Reply:
Thanks @narrieta ! Changed to warning log.